### DISCUSSION TITLE
Triggers

### DISCUSSION THREAD

**Discussion by serdardidan** [duckdb/duckdb#12562] [Ideas]
Hi,

I learned about DuckDB a few days ago. I've been reading the documentation and running tests for the last few days. A small binary file that requires no installation, great data compression, high query performance (I tested it with primitive types), collation support, SQLite and PostgreSQL support, Advanced data types and functions for them and more..

But I didn't see triggers in the documents. A parsing error occurs in the CREATE TRIGGER clause. Parser Error: syntax error at or near "TRIGGER"

DuckDB does not support triggers.. But when i write to a table in a SQLite database that has a trigger with DuckDB, the trigger working good. DuckDB supports SQLite triggers. This is great. Reading and writing SQLite databases with DuckDB is an option. But DuckDB does not use SQLite indexes. In this case, query performance decreases slightly.

I really liked DuckDB. I would start using DuckDB now if it had trigger support. Doing the work of trigger manually means more code and less performance.

Is there a specific reason why DuckDB doesn't support triggers? Or will trigger support be added later?

Best regards

**Comment by dmcmurchy** [comment]
Came hunting for Triggers information today and discovered that DuckDB doesn't return the type of `trigger` when executing the query `SELECT * FROM sqlite_master` as it only returned the `table` and `index` types from the sqlite db file I was working with. 

Opening it in DBeaver and executing the same query did return the trigger type and related information.

**Comment by serdardidan** [comment]
What I mean is different. I did some tests. You can do it too.

1- When I made an insert with DuckDb to a SQLite table that has a trigger, the trigger was triggered.

2- I think that when I query a SQLite table with DuckDb, it does not use the index. Because the query runs slowly. The same query is fast with sqlite because it uses the index.

I wrote the above, but to be honest, it is normal for Duckdb not to have triggers. None of the columnar databases I know have triggers. The data in the columns is stored by compression. This makes it very difficult to support triggers.

The purpose of DuckDb is to import data fastly and make queries fastly. And it does this job wonderfully.
